src/soilbiogeochem/SoilBiogeochemDecompCascadeMIMICSMod.F90

```diff
@@ -846,12 +855,16 @@ subroutine decomp_rates_mimics(bounds, num_soilc, filter_soilc, &
     real(r8):: spinup_geogterm_s3(bounds%begc:bounds%endc) ! geographically-varying spinup term for s3
     real(r8):: spinup_geogterm_m1(bounds%begc:bounds%endc)  ! geographically-varying spinup term for m1
     real(r8):: spinup_geogterm_m2(bounds%begc:bounds%endc)  ! geographically-varying spinup term for m2
+    real(r8):: ligninNratioAvg_local(bounds%begc:bounds%endc)  ! local column-level lignin to nitrogen ratio
+    real(r8):: annsum_npp_col_local(bounds%begc:bounds%endc)  ! local annual sum of NPP at the column level
+    real(r8):: ligninNratio(bounds%begp:bounds%endp)  ! local patch-level lignin to nitrogen ratio
+    real(r8):: annsum_npp(bounds%begp:bounds%endp)  ! local annual sum of NPP at the patch level
+    real(r8):: ligninNratioAvg_scalar  ! lignin to nitrogen ratio, scalar in column-level loop
+    real(r8):: annsum_npp_col_scalar  ! annual sum of NPP, scalar in column-level loop
 
     !-----------------------------------------------------------------------
 
     associate(                                                           &
-         ligninNratioAvg => cnveg_carbonflux_inst%ligninNratioAvg_col  , & ! Input:  [real(r8) (:)     ]  column-level lignin to nitrogen ratio
-         annsum_npp_col => cnveg_carbonflux_inst%annsum_npp_col        , & ! Input:  [real(r8) (:)     ]  annual sum of NPP at the column level (gC/m2/yr)
```

Comment:
Moved above two lines from here to inside an if statement where they get used only when not use_fates.

Reply:
Right because now there is a local version of the array used for FATES. And there is a scalar that's assigned for either fates on/or off that actually gets used. This version is only used for the non-FATES case.

Reply:
I think this is OK. Technically the local version of the array is only used when FATES is on, but will be allocated while the subroutine is active even without FATES active. But, to get around that you'd need to add explicit allocate/deallocate statements which would add more infrastructure.